packages/cli/src/deploy/deploy.ts

```diff
@@ -0,0 +1,64 @@
+import { Account, Address, Chain, Client, Transport } from "viem";
+import { ensureDeployer } from "./ensureDeployer";
+import { deployWorld } from "./deployWorld";
+import { ensureTables } from "./ensureTables";
+import { Config, ConfigInput, WorldDeploy } from "./common";
+import { ensureSystems } from "./ensureSystems";
+import { waitForTransactionReceipt } from "viem/actions";
+import { getWorldDeploy } from "./getWorldDeploy";
+import { ensureFunctions } from "./ensureFunctions";
+import { ensureModules } from "./ensureModules";
+
+type DeployOptions<configInput extends ConfigInput> = {
+  client: Client<Transport, Chain | undefined, Account>;
+  config: Config<configInput>;
+  worldAddress?: Address;
+};
+
+/** Given a viem client and MUD config, we attempt to introspect the world (or deploy a new one if no world address is provided) and do the minimal amount of work to make the world match the config (e.g. deploy new tables, replace systems, etc.) */
```

Comment:
surely vim can wrap lines if you ask it to?

we should add a linter/autoformatter if this is what we want everywhere, so that we're wrapping consistently